File: backend/Core/DTOs/PerfumeDetailsDto.cs
using Core.Enums;

namespace Core.DTOs
{
    public sealed class PerfumeDetailsDto
    {
        public int PerfumeId { get; init; }
        public string Name { get; init; } = null!;
        public string Brand { get; init; } = null!;

        public string? ImageUrl { get; init; }

        public double AvgRating { get; init; }
        public int RatingCount { get; init; }
        public int CommentCount { get; init; }

        public GenderEnum? Gender { get; init; }
        public SeasonEnum? Season { get; init; }
        public DaytimeEnum? Daytime { get; init; }
        public double? Longevity { get; init; }
        public double? Sillage { get; init; }

        public int? MyRating { get; init; }
        public string? MyReview { get; init; }

        public GenderEnum? MyGenderVote { get; init; }
        public SeasonEnum? MySeasonVote { get; init; }
        public DaytimeEnum? MyDaytimeVote { get; init; }
        public LongevityEnum? MyLongevityVote { get; init; }
        public SillageEnum? MySillageVote { get; init; }

        public IReadOnlyList<string> Groups { get; init; } = [];
        public IReadOnlyList<PerfumeNoteGroupDto> NoteGroups { get; init; } = [];

        public IReadOnlyList<ReviewDto> Reviews { get; init; } = [];
    }
}

File: mobile/src/app/models/perfumedetails.dto.ts
import { GenderEnum } from '../enums/gender.enum';
import { SeasonEnum } from '../enums/season.enum';
import { DaytimeEnum } from '../enums/daytime.enum';
import { LongevityEnum } from '../enums/longevity.enum';
import { SillageEnum } from '../enums/sillage.enum';
import { ReviewDto } from './review.dto';
import { PerfumeNoteGroupDto } from './perfumenotegroup.dto';

export interface PerfumeDetailsDto {
  perfumeId: number;
  name: string;
  brand: string;
  imageUrl?: string | null;

  avgRating: number;
  ratingCount: number;
  commentCount: number;

  gender?: GenderEnum | null;
  season?: SeasonEnum | null;
  daytime?: DaytimeEnum | null;
  longevity?: number | null;
  sillage?: number | null;

  myRating?: number | null;
  myReview?: string | null;

  myGenderVote?: GenderEnum | null;
  mySeasonVote?: SeasonEnum | null;
  myDaytimeVote?: DaytimeEnum | null;
  myLongevityVote?: LongevityEnum | null;
  mySillageVote?: SillageEnum | null;

  groups: string[];
  noteGroups: PerfumeNoteGroupDto[];
  reviews: ReviewDto[];
}
